### PR TITLE
fix: [SRM-15001]: Fix duplicate key exception while updating of compo…

### PR DESCRIPTION
**Pull request by Varsha-lalwani**
…site SLO Records

## Describe your changes
The issue is similar to what we faced in simple SLO. Duplicate records. We still don't know why we got that. But this will fix it.  Created a techdebt ticket for figuring out why it occured in the first place.
https://harness.atlassian.net/browse/SRM-15002

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49522)

### CODE REVIEW

**Comment by Varsha-lalwani** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/CompositeSLORecordServiceImpl.java [hunk 1]
```diff
@@ -227,8 +227,12 @@ public void updateCompositeSLORecords(
       String verificationTaskId, Instant startTime, Instant endTime, SLIEvaluationType sliEvaluationType) {
     List<CompositeSLORecord> toBeUpdatedSLORecords =
         getSLORecords(verificationTaskId, startTime, endTime.plus(1, ChronoUnit.MINUTES));
-    Map<Instant, CompositeSLORecord> sloRecordMap =
-        toBeUpdatedSLORecords.stream().collect(Collectors.toMap(CompositeSLORecord::getTimestamp, Function.identity()));
+    Map<Instant, CompositeSLORecord> sloRecordMap = toBeUpdatedSLORecords.stream().collect(
+        Collectors.toMap(CompositeSLORecord::getTimestamp, Function.identity(), (sloRecord1, sloRecord2) -> {
+          log.info("Duplicate SLO Key detected sloId: {}, timeStamp: {}", serviceLevelObjective.getUuid(),
+              sloRecord1.getTimestamp());
+          return sloRecord1.getLastUpdatedAt() > sloRecord2.getLastUpdatedAt() ? sloRecord1 : sloRecord2;
```
Comment (naveen-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/CompositeSLORecordServiceImpl.java:234]: You can use `BinaryOperator.maxBy` to simplify the code

**Review by naveen-harness — APPROVED** [review]

**Comment by Varsha-lalwani** [comment]
trigger messagecheck

**Review by deepak-harness — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49522)* status: :shipit: complete! ✓

**Review by deepak-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49522)* status: :shipit: complete! ✓

**Pull request merged**